api/upgradeseries/export_test.go

```diff
@@ -0,0 +1,19 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package upgradeseries
+
+import (
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/api/base"
+	"github.com/juju/juju/api/common"
+)
+
+func NewStateFromCaller(caller base.FacadeCaller, authTag names.Tag) *State {
```

Comment:
This doesn't need to be in export_text.
It's not accessing any unexported artefacts.

Reply:
This method is just for testing and access the un-exported fields of the `State` (now Client) struct. There is a constructor function but the intention of this testing function seems to be by pass some logic in the production code constructor.